code/datums/components/acid.dm

```diff
@@ -0,0 +1,206 @@
+	set_volume(acid_volume + _acid_volume)
+
+/// Sets the acid volume to a new value. Limits the acid volume by the amount allowed to exist on the parent atom.
+/datum/component/acid/proc/set_volume(new_volume)
+	acid_volume = clamp(new_volume, 0, max_volume)
+	if(!acid_volume)
+		qdel(src)
+
+
+/// Handles the slow corrosion of the parent [/atom].
+/datum/component/acid/process()
+	process_effect?.InvokeAsync()
+	set_volume(acid_volume - (ACID_DECAY_BASE + (ACID_DECAY_SCALING*round(sqrt(acid_volume)))))
+
+/// Handles processing on a [/obj].
+/datum/component/acid/proc/process_obj(obj/target)
+	if(target.resistance_flags & ACID_PROOF)
+		return
+	target.take_damage(min(1 + round(sqrt(acid_power * acid_volume)*0.3), OBJ_ACID_DAMAGE_MAX), BURN, ACID, 0)
+
+/// Handles processing on a [/mob/living].
+/datum/component/acid/proc/process_mob(mob/living/target)
+	target.acid_act(acid_power, acid_volume)
+
+/// Handles processing on a [/turf].
+/datum/component/acid/proc/process_turf(turf/target)
+	var/turf/T = parent
+	var/acid_used = min(acid_volume * 0.05, 20)
+	var/applied_targets = 0
+	for(var/A in T)
+		var/atom/acid_target = A
+		acid_target.acid_act(acid_power, acid_used)
+		applied_targets++
+
+	set_volume(acid_volume - (acid_used * applied_targets))
+
+	// Snowflake code for handling acid melting walls. TODO: Move integrity handling to the atom level so this can be desnowflaked.
+	if(acid_power < ACID_POWER_MELT_TURF)
+		return
+
+	switch(parent_integrity--)
+		if(-INFINITY to 0)
+			T.visible_message("<span class='warning'>[T] collapses under its own weight into a puddle of goop and undigested debris!</span>")
+			T.acid_melt()
+		if(0 to 4)
+			T.visible_message("<span class='warning'>[T] begins to crumble under the acid!</span>")
+		if(4 to 8)
+			T.visible_message("<span class='warning'>[T] is struggling to withstand the acid!</span>")
+		if(8 to 16)
+			T.visible_message("<span class='warning'>[T] is being melted by the acid!</span>")
+		if(16 to 24)
+			T.visible_message("<span class='warning'>[T] is holding up against the acid!</span>")
+
+
+/// Used to maintain the acid overlay on the parent [/atom].
+/datum/component/acid/proc/on_update_overlays(atom/parent_atom, list/overlays)
+	SIGNAL_HANDLER
+
+	overlays += mutable_appearance('icons/effects/acid.dmi', parent_atom.custom_acid_overlay || ACID_OVERLAY_DEFAULT)
+
+/// Alerts any examiners to the acid on the parent atom.
+/datum/component/acid/proc/on_examine(atom/A, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	examine_list += "<span class='danger'>[A.p_theyre()] covered in corrosive liquid!</span>"
+
+/// Makes it possible to clean acid off of objects.
+/datum/component/acid/proc/on_clean(atom/A, clean_types)
+	SIGNAL_HANDLER
+
+	if(!(clean_types & CLEAN_TYPE_ACID))
+		return NONE
+	qdel(src)
+	return TRUE // I know that returning booleans is bad form but this is how the proc handles it.
```

Comment:
Don't, create a flag. Check in the signal definitions for examples.

Reply:
Done, should be in the next push.

Reply:
If this proc is used elsewhere, not for signal returns, then make the signal handler call it and return a flag depending on its result.

Reply:
I meant that the proc that sends the signal handled it like a boolean.